### ISSUE TITLE
[Bug]: system.ram.used too high as it does not take into account zfs arc

### ISSUE THREAD

**Issue by callumgare** [netdata/netdata#12843]
### Bug description

The system.ram chart for one of my nodes looks like this:
![Screen Shot 2022-05-08 at 14 37 57](https://user-images.githubusercontent.com/346340/167282107-dbd70eb9-f0d5-4130-9ac7-8daa478aa8d8.png)

That looks a bit worrying because it makes it seems like I have less than 1GiB of headroom but in reality I actually have over 10GiB. The reason for this is that it doesn't take into account ZFS ARC which (in my admittedly limited understanding) is ZFS's equivalent of the buffer/cache system used for other file systems. In my case ARC is using nearly 10GiB: 
![Screen Shot 2022-05-08 at 14 39 48](https://user-images.githubusercontent.com/346340/167282252-243debbd-4ee6-4982-be14-88df14ac11fb.png)

This is also an issue with mem.available:
![Screen Shot 2022-05-08 at 14 53 07](https://user-images.githubusercontent.com/346340/167282385-42767852-f448-4b4a-a382-13bde8ae9426.png)

Because of this system.ram.used & mem.available is not a meaningful value for most ZFS users and so those charts are unhelpful (in fact they can be detrimental if you're like me and think some thing is wrong at first because I didn't realise it wasn't taking into account ARC).

This because extra annoying it triggers the "ram_avaliable" mem.available alarm when it there is no need as it based off those values:
![Screen Shot 2022-05-08 at 14 59 35](https://user-images.githubusercontent.com/346340/167282495-3507497c-cf32-41c5-b46b-9fa706108039.png)

Netdata does know about the existence of ARC as it reports on it under "ZFS Cache":
![Screen Shot 2022-05-08 at 15 03 13](https://user-images.githubusercontent.com/346340/167282647-0b0cefa8-ac5a-4603-b4d7-3fdccfeff83e.png)

Also the ram_in_use alarm seems to take it into account in all the other alarms as far as I can tell including the "ram_avaliable" alarm for system.ram:
![Screen Shot 2022-05-08 at 15 02 10](https://user-images.githubusercontent.com/346340/167282784-00982e7f-1304-47c1-9102-fbd4ac5e9ea3.png)
![Screen Shot 2022-05-08 at 15 11 09](https://user-images.githubusercontent.com/346340/167282788-a13d8149-9c4b-4f72-abb2-1ae79e0c0588.png)

### Expected behavior

As mentioned in the description other alarms deal with this by explicit removing the arc usage from system.ram.used. However I believe this issue should actually be fixed at the source and instead system.ram + mem.available should take ARC usage into account. This would mean the related charts would be meaningful and it would also mean that the memory related alerts don't need to each manually factor in ARC usage.

This could be done by either adding a new metric called something like zfs_cache or by adding that usage to either the cached or buffers metrics (I don't know enough about how ARC works to know which would make the most sense here).

That amount should be calculated by whatever ZFS ARC is currently using minus whatever ARC's minimum size usage is set to (by default 1GiB I believe). The alerts that already factor in ARC currently use this formula.

### Steps to reproduce

Unfortunately reproducing this is tricky unless you have a machine using ZFS handy which I presume someone does given that netdata already knows has the ZFS Cache metrics. If you do have that then you can check this by:
1. Running `sudo arc_summary` and checking the value for `ARC size (current):` which is curently being included in system.ram.used.

### Installation method

docker

### System info

```shell
Inside the docker container:
Linux jemima.dollhouse.zone 5.15.30-2-pve #1 SMP PVE 5.15.30-3 (Fri, 22 Apr 2022 18:08:27 +0200) x86_64 Linux
/etc/alpine-release:3.15.4
/etc/os-release:NAME="Alpine Linux"
/etc/os-release:ID=alpine
/etc/os-release:VERSION_ID=3.15.4
/etc/os-release:PRETTY_NAME="Alpine Linux v3.15"

On the host:
callumgare@jemima ~> uname -a; grep -HvE "^#|URL" /etc/*release
Linux jemima 5.15.30-2-pve #1 SMP PVE 5.15.30-3 (Fri, 22 Apr 2022 18:08:27 +0200) x86_64 GNU/Linux
/etc/os-release:PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
/etc/os-release:NAME="Debian GNU/Linux"
/etc/os-release:VERSION_ID="11"
/etc/os-release:VERSION="11 (bullseye)"
/etc/os-release:VERSION_CODENAME=bullseye
/etc/os-release:ID=debian
```

### Netdata build info

```shell
bash-5.1# netdata -W buildinfo
Version: netdata v1.34.0-120-nightly
```

### Additional info

Looking at other programs `htop` correctly takes ARC into account:
![Screen Shot 2022-05-08 at 14 38 13](https://user-images.githubusercontent.com/346340/167282327-db5bf88a-11ec-4924-b04f-44709935a866.png)

However some older programs like `top` and `free` don't appear to:
![Screen Shot 2022-05-08 at 13 12 45](https://user-images.githubusercontent.com/346340/167282338-45e86b6a-d4d2-4863-876f-39e9f00d5f7d.png)
![Screen Shot 2022-05-08 at 13 12 54](https://user-images.githubusercontent.com/346340/167282339-31ebcc90-9bac-4143-aefc-8d361c5f90c5.png)

**Comment by callumgare** [comment]
While this doesn't fix the issue of the charts showing misleading values, to anyone else in this situation a workaround to  least fix the mem.avaliable "ram_avaliable" alert is to cd'ing into your netdata config dir, run `sudo ./edit-config health.d/ram.conf`, and change the `calc` value for the mem.avaliable's "ram_avaliable" alert from it's original value of `$avail * 100 / ($system.ram.used + $system.ram.cached + $system.ram.free + $system.ram.buffers)` to `($avail + $system.ram.used_ram_to_ignore) * 100 / ($system.ram.used + $system.ram.cached + $system.ram.free + $system.ram.buffers)`.

**Comment by ilyam8** [comment]
Hi, @callumgare. Thanks for the detailed bug report 👍 

The `mem.available` chart data is `MemAvailable` from `/proc/meminfo`.

It is:

- from `man procfs`.


> **MemAvailable** %lu (since Linux 3.14)
>  An estimate of how much memory is available for starting new applications, without swapping.


- from `man free`.


> **available**
>   Estimation of how much memory is available for starting new applications, without swapping. Unlike the data provided by the cache or free fields,  this field takes into account the page cache and also that not all reclaimable memory slabs will be reclaimed due to items being in use (MemAvailable in /proc/meminfo, available on kernels 3.14, emulated on kernels 2.6.27+, otherwise the same as free).

---

I think we don't modify this value because we show the kernel estimation: if the kernel believes that X memory can be used without swapping that it is what it is.

> Looking at other programs htop correctly takes ARC into account:
> However some older programs like top and free don't appear to:

That is an interesting topic, but we get different and mutually exclusive requests: some people want these charts to accurately reflect `free -h` because they have been using this command for years, and getting different values is confusing.

---

`htop` [does the following](https://github.com/htop-dev/htop/pull/180/files):

 - usedMem -= ZFS ARC size
 - cachedMem += ZFS ARC size

I guess we can do the same for `system.ram`, not sure about `mem.available`. @Ferroin what do you think?

**Comment by ilyam8** [comment]
@callumgare can you test #12847?

**Comment by callumgare** [comment]
Wow! What incredible turn around time 🤩

I built it with `sudo ./netdata-installer.sh` first on the master branch and then on fix_proc-meminfo_consider_zfs_arc_as_cache to compare (hence the chart values changing dramatically towards the end) and it does indeed seem to be working.
![Screen Shot 2022-05-08 at 23 33 45](https://user-images.githubusercontent.com/346340/167298786-83c09b3a-3e24-4293-820e-3ac59a85df87.png)

![Screen Shot 2022-05-08 at 23 34 34](https://user-images.githubusercontent.com/346340/167298825-d426cdf9-c5e5-425f-8b91-bd5b5053b39c.png)

Amazing, thanks for that ❤️

**Comment by ilyam8** [comment]
@callumgare I think `used -= (arcsize - arc_min)` is a better way to calc used than `used -= arcsize`, I guess we can do a patch to `htop` 🤔

**Comment by callumgare** [comment]
@ilyam8 Yeah, that makes sense to me. Good find tracking down the htop patch.
